File: app/src/main/java/com/abaffym/moviesapp/data/remote/rest/MoviesApi.java
package com.abaffym.moviesapp.data.remote.rest;

import com.abaffym.moviesapp.data.remote.entity.DiscoverMoviesResponse;
import com.abaffym.moviesapp.data.remote.entity.MovieEntity;

import io.reactivex.Single;
import retrofit2.http.GET;

public interface MoviesApi {

	@GET("discover/movie")
	Single<DiscoverMoviesResponse<MovieEntity>> discoverMovies();

}

File: app/src/main/java/com/abaffym/moviesapp/data/remote/rest/MoviesApi.kt
package com.abaffym.moviesapp.data.remote.rest

import com.abaffym.moviesapp.data.remote.entity.DiscoverMoviesResponse
import com.abaffym.moviesapp.data.remote.entity.MovieEntity

import io.reactivex.Single
import retrofit2.http.GET

interface MoviesApi {

    @GET("discover/movie")
    fun discoverMovies(): Single<DiscoverMoviesResponse<MovieEntity>>

}
